src/main/java/it/pagopa/transactions/client/NotificationsServiceClient.java

```diff
@@ -0,0 +1,28 @@
+package it.pagopa.transactions.client;
+
+import it.pagopa.generated.notifications.v1.api.DefaultApi;
+import it.pagopa.generated.notifications.v1.dto.NotificationEmailRequestDto;
+import it.pagopa.generated.notifications.v1.dto.NotificationEmailResponseDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Mono;
+
+import java.util.Map;
+
+@Component
+public class NotificationsServiceClient {
+    @Value("${notificationsService.apiKey}")
+    private String notificationsServiceApiKey;
+
+    @Autowired
+    private DefaultApi notificationsServiceApi;
+
+    public Mono<NotificationEmailResponseDto> sendNotificationEmail(NotificationEmailRequestDto notificationEmailRequestDto) {
+        if (!(notificationEmailRequestDto.getParameters() instanceof Map)) {
+            return Mono.error(new IllegalArgumentException("Notifications service `parameters` field in `sendNotificationsEmail` request body must implement `java.util.Map`"));
```

Comment:
Do you think we can check better which kind of parameters come from dto with a custom swagger in which only ecommerce parameter type are listed?

Reply:
Yes, better type checking will come from another PR when we'll integrate with JSON Schemas provided by notifications-service templates 😃